### DISCUSSION TITLE
Option to download chat logs\save them locally

### DISCUSSION THREAD

**Discussion by CinderBH** [stoatchat/discussions#580] [Feature Suggestions]
One thing I always sorely missed from the transition from early chat clients like mIRC, AIM and MSN to Skype and everything that came after is the ability to download chat logs or, even better, save them locally in real time. In this case, maybe with the option to skip pictures\media to keep file size under control.
It always frustrates me how all my important interactions with friends are held hostage in a server and I'm liable to lose them all if anything happens to my account. In fact, Discord is particularly hostile to people attempting to download their logs, and it's possibly the reason why a lot of people are still anchored to it. Meanwhile with AIM that was never a concern, and even when it was shut down for good it didn't sting so much because I still had all my logs.

For one, the Element client already allows downloading Matrix logs, so maybe it shouldn't be too difficult to implement? I figure the only issue would be people who don't consent to having their messages logged, but perhaps something could be done to give them the option to opt out.

**Comment by kittieluce** [comment]
this all the way

**Reply by LugiaLucarioArceus** [reply]
Agreed.

**Reply by SpookRooster** [reply]
100% agree. exporting logs with the option of files included would be amazing.
